src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3207,24 +3239,31 @@ void ViewProviderSketch::camSensCB(void *data, SoSensor *)
     auto vp = proxyVPrdr->vp;
     auto cam = proxyVPrdr->renderMgr->getCamera();
 
-    auto rotSk = Base::Rotation(vp->getDocument()->getEditingTransform()); //sketch orientation
+    vp->onCameraChanged(cam);
+}
+
+void ViewProviderSketch::onCameraChanged(SoCamera* cam)
+{
+    auto rotSk = Base::Rotation(getDocument()->getEditingTransform()); //sketch orientation
     auto rotc = cam->orientation.getValue().getValue();
     auto rotCam = Base::Rotation(rotc[0], rotc[1], rotc[2], rotc[3]); // camera orientation (needed because float to double conversion)
 
     // Is camera in the same hemisphere as positive sketch normal ?
-    auto orientation = (rotCam.invert()*rotSk).multVec(Base::Vector3d(0,0,1));
-    auto tmpFactor = orientation.z<0?-1:1;
+    auto orientation = (rotCam.invert() * rotSk).multVec(Base::Vector3d(0, 0, 1));
+    auto tmpFactor = orientation.z < 0 ? -1 : 1;
 
-    if (tmpFactor != vp->viewOrientationFactor) { // redraw only if viewing side changed
+    if (tmpFactor != viewOrientationFactor) { // redraw only if viewing side changed
         Base::Console().Log("Switching side, now %s, redrawing\n", tmpFactor < 0 ? "back" : "front");
-        vp->viewOrientationFactor = tmpFactor;
-        vp->draw();
+        viewOrientationFactor = tmpFactor;
+        draw();
 
         QString cmdStr = QStringLiteral(
             "ActiveSketch.ViewObject.TempoVis.sketchClipPlane(ActiveSketch, ActiveSketch.ViewObject.SectionView, %1)\n")
-            .arg(tmpFactor<0?QLatin1String("True"):QLatin1String("False"));
+            .arg(tmpFactor < 0 ? QLatin1String("True") : QLatin1String("False"));
         Base::Interpreter().runStringObject(cmdStr.toLatin1());
     }
+
+    drawGrid(true);
```

Comment:
This can be called very often on fast rotating the view. We have to check execution time and see if counter-measures are needed. ;)

Reply:
Yes. We keep an eye.